### ISSUE TITLE
Support for ASP.NET Core 3.0

### ISSUE THREAD

**Issue by martincostello** [aspnet-contrib/AspNet.Security.OAuth.Providers#324]
Support for ASP.NET Core 3.0 has been started with support for preview 6 in #280, which has been merged into the [`dev-3.0.0`](https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/tree/dev-3.0.0) branch.

Prerelease packages are available for testing from the [MyGet feed](https://www.myget.org/gallery/aspnet-contrib).

**Comment by martincostello** [comment]
Provider packages that support ASP.NET Core 3.0 preview 7 (`3.0.100-preview7-012821`) are now available in the MyGet feed starting at version `3.0.0-preview-0856`.

**Comment by martincostello** [comment]
Provider packages that support ASP.NET Core 3.0 preview 8 (`3.0.100-preview8-013656`) are now available in the MyGet feed starting at version `3.0.0-preview-0858`.

**Comment by martincostello** [comment]
Provider packages that support ASP.NET Core 3.0 preview 9 (`3.0.100-preview9-014004`) are now available in the MyGet feed starting at version `3.0.0-preview9.19455.71`.

**Comment by mscrivo** [comment]
Any reason not to publish these to nuget as well?  That would be more inline with what MS is doing with .net core preview packages and would help those of us testing out the preview releases with a bit less friction.

**Comment by martincostello** [comment]
It’s just a bit more manual work to curate that, rather than just have a CI feed where we don’t need to worry about “clogging it up” with lots of versions from CI.

**Comment by martincostello** [comment]
Provider packages that support ASP.NET Core 3.0 RC1 (`3.0.100-rc1-014190`) are now available in the MyGet feed starting at version `3.0.0-rc1.19466.76`.

**Comment by martincostello** [comment]
The `dev-3.0.0` branch has now been merged to `dev`, and then to `rel/3.0.0` for the final release.

I'm on vacation for the next two weeks now, but packages for the providers compiled against ASP.NET Core 3.0.0 GA and the .NET Core 3.0.100 SDK are available in [MyGet](https://www.myget.org/gallery/aspnet-contrib) starting from `3.0.0-release.19473.83`.

**Comment by Herdo** [comment]
The release when you're back is a full release, no pre-release, right?

**Comment by martincostello** [comment]
That’s the idea, though the only difference between the two is likely to be the version number.

**Comment by mscrivo** [comment]
> That’s the idea, though the only difference between the two is likely to be the version number.

Well that, and that it'll be available on nuget :) .. No rush though, enjoy your vacation!

**Comment by martincostello** [comment]
Candidate `3.0.0` packages have [been built](https://ci.appveyor.com/project/aspnet-contrib/aspnet-security-oauth-providers/builds/27889663/artifacts) and are in [MyGet](https://www.myget.org/gallery/aspnet-contrib) and GitHub release notes prepped.

**Comment by martincostello** [comment]
I've validated the `3.0.0` packages for Amazon and Apple with some sites of my own.

**Comment by mscrivo** [comment]
Any ETA yet on nuget publishing?  This is literally the last piece we're waiting on before we can upgrade.

**Comment by kevinchalet** [comment]
This week. It's at the top of my list of things to do 😄

**Comment by martincostello** [comment]
I don't have access to publish to NuGet.org, so it has to wait until Kévin has the time to do it.

To unblock yourself, I'd recommend using the packages out of MyGet until then.

**Comment by kevinchalet** [comment]
Note: the release of the 3.0.0 packages will be delayed a bit, as an issue was identified with the way we strong name our assemblies (it seems delayed or public signing is enforced by Arcade by default).

While it's not critical and doesn't prevent using the packages - that are .NET Core-only, where it matters less - we'll want to fix that nevertheless.

Edit: it's even worse than that: our assemblies are currently "public signed" by the MicrosoftShared key 😄

**Comment by kevinchalet** [comment]
OpenIddict is impacted by the same bug (https://github.com/openiddict/openiddict-core/pull/827 should fix it). I'll port the same fix to the OAuth 2.0/OpenID 2.0 repos tomorrow.

**Comment by kevinchalet** [comment]
The 3.0.0 packages are now on NuGet.org 🎉
